Add a new configuration item: proxy.http.port

PowderLi opened the pull request:
/kind bug
fix #25873
bring by #24761

jaime0815 commented on internal/distributed/proxy/service.go:
```diff
@@ -132,6 +133,10 @@ func authenticate(c *gin.Context) {
 			return
 		}
 	}
+	if !proxy.Params.CommonCfg.AuthorizationEnabled {
```
1. Why need to set root as a default user if authorization is disabled? 
2.  is it reasonable to set the current name to root after parsing the user name successfully if authorization is disabled

jaime0815 commented on internal/distributed/proxy/service.go:
```diff
@@ -407,13 +412,67 @@ func (s *Server) init() error {
+				creds, err := tls.LoadX509KeyPair(Params.ServerPemPath, Params.ServerKeyPath)
+				if err != nil {
+					log.Error("proxy can't create creds", zap.Error(err))
+					log.Error("proxy can't create creds", zap.Error(err))
```
duplicated log

jaime0815 commented on internal/distributed/proxy/httpserver/handler_v1.go:
```diff
@@ -619,11 +633,13 @@ func (h *Handlers) search(c *gin.Context) {
-	if authErr != nil {
-		c.JSON(http.StatusUnauthorized, gin.H{HTTPReturnCode: Code(authErr), HTTPReturnMessage: authErr.Error()})
-		return
+	if proxy.Params.CommonCfg.AuthorizationEnabled {
```
we'd better add a new checkAuthorization method to reduce duplicated code

PowderLi commented:
rerun ut

jaime0815 commented on internal/distributed/proxy/httpserver/handler_v1.go:
```diff
@@ -47,7 +47,7 @@ func (h *Handlers) describeCollection(c *gin.Context, dbName string, collectionN
 		CollectionName: collectionName,
 	}
-	if needAuth {
+	if needAuth && proxy.Params.CommonCfg.AuthorizationEnabled {
```
Why needAuth is true if the requested prefix is /vector/collections/describe? yet other requests don't need to check auth.

jaime0815 commented on internal/distributed/proxy/httpserver/handler_v1.go:
```diff
@@ -190,11 +192,13 @@ func (h *Handlers) createCollection(c *gin.Context) {
-	if authErr != nil {
-		c.JSON(http.StatusUnauthorized, gin.H{HTTPReturnCode: Code(authErr), HTTPReturnMessage: authErr.Error()})
-		return
+	if proxy.Params.CommonCfg.AuthorizationEnabled {
```
Suggest using a method to check auth to reduce duplicated code.

jaime0815 commented:
To ensure that GRPC and restful pass username as the requested tenant ID through context,  we still need to process[ this method](https://github.com/milvus-io/milvus/blob/0ce6dda785ab9112ff98ddaec394e11642b9e19c/internal/proxy/util.go#L818)  to make it work well if it is a restful request.

jaime0815 commented:
/run-cpu-e2e

PowderLi committed:
Add a new configuration item: proxy.http.port
    user define the port for high-level restful api by themselves
    user can only use the default port same as grcp only while tlsMode == 0
fix need authenticate only while authorizationEnable: true

Signed-off-by: PowderLi <min.li@zilliz.com>

PowderLi commented:
duplicated #26008
